src/main/java/net/minecraftforge/items/CapabilityItemHandler.java

```diff
@@ -0,0 +1,84 @@
+package net.minecraftforge.items;
+
+import net.minecraft.block.Block;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.inventory.ISidedInventory;
+import net.minecraft.inventory.InventoryLargeChest;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTBase;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.tileentity.TileEntityChest;
+import net.minecraft.util.BlockPos;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.CapabilityInject;
+import net.minecraftforge.common.capabilities.CapabilityManager;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import net.minecraftforge.event.AttachCapabilitiesEvent;
+import net.minecraftforge.fml.common.eventhandler.EventPriority;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.items.wrapper.InvWrapper;
+import net.minecraftforge.items.wrapper.SidedInvWrapper;
+
+import java.util.concurrent.Callable;
+
+public class CapabilityItemHandler {
+    @CapabilityInject(IItemHandler.class)
+    public static Capability<IItemHandler> ITEM_HANDLER_CAPABILITY = null;
+    
+    static
+    {
+        CapabilityManager.INSTANCE.register(IItemHandler.class, new Capability.IStorage<IItemHandler>() {
+            @Override
+            public NBTBase writeNBT(Capability<IItemHandler> capability, IItemHandler instance, EnumFacing side)
+            {
+                NBTTagList nbtTagList = new NBTTagList();
+                int size = instance.getSlots();
+                for (int i = 0; i < size; i++)
+                {
+                    ItemStack stack = instance.getStackInSlot(i);
+                    if (stack != null)
+                    {
+                        NBTTagCompound itemTag = new NBTTagCompound();
+                        if (size < 256)
+                            itemTag.setByte("Slot", (byte) i);
+                        else
+                            itemTag.setInteger("Slot", i);
+                        stack.writeToNBT(itemTag);
+                        nbtTagList.appendTag(itemTag);
+                    }
+                }
+                return nbtTagList;
+            }
+
+            @Override
+            public void readNBT(Capability<IItemHandler> capability, IItemHandler instance, EnumFacing side, NBTBase base)
+            {
+                IItemHandlerModifiable itemHandlerModifiable = (IItemHandlerModifiable) instance;
+                NBTTagList tagList = (NBTTagList) base;
```

Comment:
Throw a proper exception if you need something more then the interface, this will error out on class cast which isnt as useful in debug.